pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtimetoofflinesegments/RealtimeToOfflineSegmentsTaskGenerator.java

```diff
@@ -204,6 +205,13 @@ public String getTaskType() {
       if (segmentNames.isEmpty() || skipGenerate) {
         LOGGER.info("Found no eligible segments for task: {} with window [{} - {}). Skipping task generation", taskType,
             windowStartMs, windowEndMs);
+        if (!skipGenerate) {
```

Comment:
can we go back to line 164 in this iteration itself, if we detect this? So that we dont have to wait for the next scheduling?